### ISSUE TITLE
pandas: selecting rows does not work

### ISSUE THREAD

**Issue by azjps** [saulpw/visidata#266]
To reproduce, open any `PandasSheet` and try pressing `s`; no rows appear as selected.

The problem comes down to that `selection.py` heavily uses `id()` to track rows, and it will either use `__iter__()` or `__getitem__()` to access the rows. However, with `pandas`, these two methods yield different objects with different ids:
```python
$ ipython
Python 3.7.2 (default, Dec 29 2018, 06:19:36)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.3.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import pandas as pd

In [2]: df = pd.DataFrame({"a": [1, 2], "b": [3, 4]})

In [3]: [id(r) for r in df]
Out[3]: [140290703622920, 140290703910032]

In [4]: id(df.iloc[0]), id(df.iloc[1])
Out[4]: (140290637990936, 140290637990936)
```
In general, `id()` is not reliable for `pandas` rows as for heterogeneous-dtype dataframes a copy will be made when accessing rows (and probably also for some homogeneous-dtype):
```python
In [5]: df = pd.DataFrame({"a": [1, 2], "b": ["c", "d"]})

In [6]: df
Out[6]:
   a  b
0  1  c
1  2  d

In [7]: id(df.iloc[0])
Out[7]: 140290624720968

In [8]: id(df.iloc[0])
Out[8]: 140290663594864
```
I think the right way to work around this would be to allow customizing how to define an id for a row. For pandas, we can usually use the row label assuming a default `Int64Index`. If the DataFrame has duplicated entries in its index then its a bit hopeless without significant refactoring.

**Comment by azjps** [comment]
Actually I take that back, it would probably make more sense to override all of the selection-related API and instead use a boolean mask instead to represent the selected rows. That would avoid any issues with working with individual rows or duplicate index entries. To do so though we would need to pass the integer index of the row as opposed to the row itself into `select()`; this could either be done by passing the information through `DataFrameAdapter.__getitem__` or modifying the keybinding, and I'm not sure which option makes more sense.

**Comment by azjps** [comment]
Closed by #267
